### PR TITLE
Add PHAR file to GitHub releases (PHIVE support)

### LINKED ISSUE #489: Make it installable by phive
[Phive](https://phar.io/) is useful tool for managing .phar archives. The only thing your will need to do is sign future releases with GPG key. The instructions you can find at the Phive's site (see bottom of a page).

### PR DESCRIPTION
**Pull request by filips123**
Add support for **signed** PHIVE installation. Fix #489.

It will not be available until new release, because PHAR will be added to it.

Also, @bobthecow must set up GPG keys and add them to keyservers so PHAR will be signed. Details how to do that are [on PHIVE site](https://phar.io/howto/sign-and-upload-to-github.html) (section prerequisites) and [in this article](https://andreas.heigl.org/2017/01/19/encrypt-a-build-result-automaticaly/) (all about creating and encrypting keys and adding them to Travis CI environment vars).

Basically, encrypted public and private keys should be added to `.github/keys.asc` in this repository, key ID should be added to Travis CI environment variable `KEY_ID`, decryption key should be added to environment variable `DECRYPT_KEY` and signing key should be added to environment variable `SIGN_KEY`.

This will enable installation with:

```bash
phive install bobthecow/psysh
```

After that, pull request to PHIVE should be added to make alias available.

### CODE REVIEW

**Comment by filips123** [comment]
@bobthecow Can you merge this PR?

File: Makefile [hunk 1]
```diff
@@ -95,13 +95,31 @@ build/psysh-php54-compat: $(PSYSH_SRC) $(PSYSH_SRC_FILES)
 build/%/psysh: vendor/bin/box build/%
 	vendor/bin/box compile --working-dir $(dir $@)
 
+# Signatures
+
+build/%/psysh.asc: build/%/psysh
```
Comment (theofidry) [inline Makefile]: and how you deal with people who want to build it for testing purposes?

Comment (filips123) [reply Makefile]: I'm not an expert in `Makefile`. do you know how could this be done to not require signing every time?

Currently, the signature is only required when making `dist`. When making `build`, only PHARs are made.

Comment (theofidry) [reply Makefile]: Ah I see, maybe that's fine then

**Comment by filips123** [comment]
@bobthecow Can you check this?

File: Makefile [hunk 2]
```diff
@@ -95,13 +95,31 @@ build/psysh-php54-compat: $(PSYSH_SRC) $(PSYSH_SRC_FILES)
...
+	gpg --fingerprint --with-colons $$KEY_ID |\
+		grep "^fpr" |\
+		sed -n 's/^fpr:::::::::\([[:alnum:]]\+\):/\1/p' |\
+		xargs gpg --batch --delete-secret-keys
+	rm -rf .github/keys.asc
 
 # Dist packages
 
-dist/psysh-$(VERSION).tar.gz: build/psysh/psysh
+dist/psysh-$(VERSION).tar.gz: build/psysh/psysh build/psysh/psysh.asc
```
[the remark this answers is not in the document]
Comment (filips123) [reply Makefile]: Distribution `.tag.gz` will contain signature so that's why it depends on it. Do you know any other solution?

File: Makefile [hunk 3]
```diff
@@ -95,13 +95,31 @@ build/psysh-php54-compat: $(PSYSH_SRC) $(PSYSH_SRC_FILES)
...
 	@mkdir -p $(@D)
 	tar -C $(dir $<) -czf $@ $(notdir $<)
 
-dist/psysh-$(VERSION)-%.tar.gz: build/psysh-%/psysh
+dist/psysh-$(VERSION)-%.tar.gz: build/psysh-%/psysh build/psysh-%/psysh.asc
```
[the remark this answers is not in the document]
Comment (filips123) [reply Makefile]: I added them because this will allow users to manually very signatures of compat versions if they will want.

File: .travis.yml [hunk 4]
```diff
@@ -32,14 +32,17 @@ script:
 after_success:
     - bash <(curl -s https://codecov.io/bash)
 
-before_deploy: make dist -j 4
+before_deploy: make dist -j 4 && cp build/psysh/psysh dist/psysh.phar && cp build/psysh/psysh.asc dist/psysh.asc
```
[the remark this answers is not in the document]
Comment (filips123) [reply .travis.yml]: As files already exist in `build/psysh/psysh.asc` and `build/psysh/psysh.asc`, this (copying) is actually not needed, as I can directly release them to GitHub Releases.

[the remark this answers is not in the document]
Comment (filips123) [reply .travis.yml]: This is now fixed.

[the remark this answers is not in the document]
Comment (bobthecow) [reply .travis.yml]: Why does it need to be named psysh.phar? I’d really rather not.

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (bobthecow) [reply Makefile]: Oh, in that case never mind. I thought it was only the top level file.

[the remark this answers is not in the document]
Comment (bobthecow) [reply Makefile]: Oh, if it’s going in the tgz this is a good spot for it :)

File: Makefile (hunk 1 above)
Comment (bobthecow) [reply Makefile]: Yeah, this will only run for the `dist` target, so it’s where you’d put it.

File: .travis.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (filips123) [reply .travis.yml]: Because PHIVE requires that.

Why you don't want this? In case of installation with PHIVE, `.phar` will be dropped so user when saving wouldn't notice it.

[the remark this answers is not in the document]
Comment (bobthecow) [reply .travis.yml]: For everyone who looks at the github release and _doesn’t_ use phive. It means the most downloadable looking thing in the release is a `.phar`, which is bound to cause confusion. I’ve never loved this about phive. I feel like it forces suboptimal experiences for non-phive users, unless I want to maintain my own repo server.

[the remark this answers is not in the document]
Comment (filips123) [reply .travis.yml]: But is this actually a problem? Because your website already shows installation with `wget` and `composer`. And if users need compat versions, they will probably already know that.

However, if you don't want that, you can set up [custom repository server](https://phar.io/distribute-your-own.html). I think that you can just use GitHub Pages for this with some CI script that automatically updates them at the new release.  If you think this would be better, I can try to create new PR for it.

**Comment by theofidry** [comment]
Might be worth checking it can be changed under PHAR? Alternatively you
could ship both files: one for phive and the other .phar; a user will not
download all the assets so I don’t thinking’s this should be a problem
